javalib/src/main/scala/java/util/StringTokenizer.scala

```diff
@@ -40,20 +42,28 @@ class StringTokenizer(
   def hasMoreElements(): Boolean = hasMoreTokens()
 
   def hasMoreTokens(): Boolean = {
-    if (delimiters == null) {
+    if (delimiters == null)
       throw new NullPointerException()
-    }
-    val length = string.length
-    if (position < length) {
-      if (returnDelimiters)
-        return true
 
-      for (i <- position until length) {
-        if (delimiters.indexOf(string.charAt(i), 0) == -1)
-          return true
+    @tailrec
+    def hasNonDelim(pos: Int, len: Int): Boolean = {
+      if (pos == len) {
+        false
+      } else if (delimiters.indexOf(string.charAt(pos), 0) == -1) {
+        true
+      } else {
+        hasNonDelim(pos + 1, len)
       }
```

Comment:
Maybe we can skip the braces to make it more readable? Similarly in the invocation below

So for example 
```suggestion
    @tailrec def hasNonDelim(pos: Int, len: Int): Boolean = {
    if (pos == len) false
    else if (delimiters.indexOf(string.charAt(pos), 0) == -1) true
    else  hasNonDelim(pos + 1, len)
```
Surprisingly compiler allows for tailrec for the rhs of `||` so another option might look like:
```
    @tailrec def hasNonDelim(pos: Int, len: Int): Boolean = {
    if (pos == len) false
    else {
       delimiters.indexOf(string.charAt(pos), 0) == -1) ||
       hasNonDelim(pos + 1, len)
     }
```

Reply:
I favor your version because it is a bit easier to reason about but the trailing recursive function after the `||` is pretty cool.

Reply:
> || is pretty cool.

Agreed.